tests/unit/test_views.py

```diff
@@ -377,6 +383,161 @@ def testDatasetsSearch(self):
         datasets = list(responseData.datasets)
         self.assertEqual(self.datasetId, datasets[0].id)
 
+    def testGenotypePhenotypeSearchBadRequest(self):
+        """
+        If nothing is provided to search on,
+        the system should return `bad request`
+        """
+        request = protocol.SearchGenotypePhenotypeRequest()
+        response = self.sendPostRequest('/genotypephenotype/search', request)
+        self.assertEqual(501, response.status_code)
+
     def testNoAuthentication(self):
         path = '/oauth2callback'
         self.assertEqual(501, self.app.get(path).status_code)
+
+    def testGenotypePhenotypeSearchFeature(self):
```

Comment:
We should break these tests into a seperate module, tests/unit/test_g2p.py. This module is really just for top-level HTTP stuff.